src/var.c

```diff
@@ -53,8 +55,37 @@ struct fi_var_t {
 
 static struct fi_var_t *fi_vars = NULL;
 
+__attribute__((visibility ("default")))
+int DEFAULT_SYMVER_PRE(fi_getvars)(struct fi_env **vars, int *count) {
+
+	struct fi_env *vhead;
+	int len = 0;
+
+	// just get a count
+	for (struct fi_var_t *ptr = fi_vars; ptr ;ptr = ptr->next, ++len) {}
+
+	vhead = calloc(len, sizeof (*vhead));
+
+	if (!vhead)
+		return -FI_ENOMEM;
+
+	int i = 0;
```

Comment:
replace len and i with just i - please don't declare variables in the middle of functions, it makes locating their declaration difficult